### PR TITLE
feat: implement first version of uml

### PR DESCRIPTION
**Pull request by JosuePrado**

### CODE REVIEW

File: docs/APi.puml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  
+}
+
+package "Persistence" {
+  
+  class DbInitializer {
+    + Initialize(String) : Void
+  }
+  
+  class DbConstance {
```
Comment (zrock-dev) [inline docs/APi.puml]: There is a typo

File: docs/APi.puml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  }
+  
+  class UserItems {
+    + UserId : Guid
+    + Expenses : [Expense]
+    + Incomes : [Income] 
+    + ExpensesCategory : [ExpensesCategory]
+  }
+  
+  class UserGoals {
```
Comment (zrock-dev) [inline docs/APi.puml:42]: I couldn't find who is using this class

File: docs/APi.puml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  }
+  
+  class Income {
+   - id : Guid
+   - amount : Decimal
+   - date : Date 
+   - source : String
+  }
+  
+  class UserItems {
```
Comment (zrock-dev) [inline docs/APi.puml:35]: I couldn't find who is using this class

File: docs/APi.puml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  
+  class MonthlyBudgeting {
+    + Context : ApplicationDbContext
+    + VerifyBudgetingStatus() : bool
+  }
+  
+  
+  ExpenseCategoryService --* ExpensesService : has a
+  ExpensesService --|> BaseService  : Implement
+  IncomeService --|> BaseService  : Implement
```
Comment (zrock-dev) [inline docs/APi.puml]: Given that BaseService is an interface, these should be realization relationships.

**Review by zrock-dev — CHANGES_REQUESTED** [review]
What's the difference between WebApplicationExtension and DbInitializer

File: docs/APi.puml (hunk 1 above)
Comment (JosuePrado) [reply docs/APi.puml]: that's right, but it's been fixed

File: docs/APi.puml (hunk 2 above)
Comment (JosuePrado) [reply docs/APi.puml:42]: this class is used by ApplicationDbContext

File: docs/APi.puml (hunk 3 above)
Comment (JosuePrado) [reply docs/APi.puml:35]: this class is used by ApplicationDbContext

File: docs/APi.puml (hunk 4 above)
Comment (JosuePrado) [reply docs/APi.puml]: Yes, it's fixed

**Comment by JosuePrado** [comment]
> WebApplicationExtension and DbInitializer

the DbInitializer is in charge of the verification, migration and initialization of the database while WebApplicationExtension is in charge of connecting the DbInitializer with the API

**Review by zrock-dev — APPROVED** [review]
Good job

**Review by diego-centella-jalafund — APPROVED** [review]

**Review by zrock-dev — APPROVED** [review]
LGTM

**Comment by zrock-dev** [comment]
I had requested a review from @diego-centella-jalafund because the PR required a review on the latest push, but it seems that only mine was required. I'll go ahead and merge the PR

**Pull request merged**